### PR TITLE
More explicit types in examples

### PR DESCRIPTION
**Pull request by armanbilge**
And some other minor tidying.

Closes https://github.com/armanbilge/calico/issues/189. @hejfelix were these the ones you were thinking, any others I should add to?

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -66,31 +66,33 @@ import cats.effect.*
...
-  div(
-    label("Your name: "),
-    input.withSelf { self =>
-      (
-        placeholder := "Enter your name here",
-        // here, input events are run through the given Pipe
-        // this starts background fibers within the lifecycle of the <input> element
-        onInput --> (_.foreach(_ => self.value.get.flatMap(name.set)))
+val component: Resource[IO, HtmlDivElement[IO]] =
+  SignallingRef[IO].of("world").toResource.flatMap { name =>
```
Comment (hejfelix) [inline docs/index.md:72]: Why linebreak?

**Comment by hejfelix** [comment]
Yeah. I think adding them to all the *.md code would hammer home the point that components are `Resource[F, HtmlElement[F]]`.

File: docs/counter.md [hunk 2]
```diff
@@ -48,10 +49,10 @@ def Counter(label: String, initialStep: Int) =
...
   }
 
-val app = div(
+val app: Resource[IO, HtmlDivElement[IO]] = div(
   h1("Let's count!"),
   Counter("Sheep", initialStep = 3)
 )
 
-app.renderInto(node.asInstanceOf[fs2.dom.Node[IO]]).allocated.unsafeRunAndForget()
+app.renderInto(node.asInstanceOf[fs2.dom.Node[IO]]).useForever.unsafeRunAndForget()
```
Comment (2chilled) [inline docs/counter.md:57]: This `asInstanceOf` looks a bit "hacky", at least for people not knowing background story. It also kind of leaks the opaque type. Maybe we can hide that somehow.

**Review by 2chilled — APPROVED** [review]

File: docs/counter.md (hunk 2 above)
Comment (armanbilge) [reply docs/counter.md:57]: Yeah, the `unsafeRunAndForget()` also is not so great 😅 maybe we can move these out of the code block 🤔

**Pull request merged**